model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -37,15 +37,21 @@ import "google/protobuf/timestamp.proto";
 message BeamConstants {
   enum Constants {
     // All timestamps in milliseconds since Jan 1, 1970.
+
+    // All timestamps of elements or window boundaries must be within
+    // the interval [MIN_TIMESTAMP_MILLIS, MAX_TIMESTAMP_MILLIS].
+    // The smallest representable timestamp of an element or a window boundary.
     MIN_TIMESTAMP_MILLIS = 0 [(beam_constant) = "-9223372036854775"];
+    // The largest representable timestamp of an element or a window boundary.
     MAX_TIMESTAMP_MILLIS = 1 [(beam_constant) =  "9223372036854775"];
+
     // The maximum timestamp for the global window.
-    // Triggers use maxTimestamp to set timers' timestamp. Timers fires when
+    // Triggers use max timestamp to set timers' timestamp. Timers fire when
     // the watermark passes their timestamps. So, the timestamp needs to be
     // smaller than the MAX_TIMESTAMP_MILLIS.
     // One standard day is subtracted from MAX_TIMESTAMP_MILLIS to make sure
-    // the maxTimestamp is smaller than MAX_TIMESTAMP_MILLIS even after rounding up
-    // to seconds or minutes. See also GlobalWindow in the Java SDK.
+    // the max timestamp is smaller than MAX_TIMESTAMP_MILLIS even after rounding up
+    // to seconds or minutes.
     GLOBAL_WINDOW_MAX_TIMESTAMP_MILLIS = 2 [(beam_constant) = "9223371950454775"];
```

Comment:
I'm curious, what's the reason to define all constants as strings?

Reply:
We would need to define an EnumValueOption for each type we would like to support.